Build failure: TypeFilter cannot be resolved to a type

I'm trying to build on MacOS and I'm having the following compilation failure:

```
[ERROR] Failed to execute goal org.eclipse.tycho:tycho-compiler-plugin:4.0.3:compile (default-compile) on project com.microsoft.java.debug.plugin: Compilation failure: Compilation failure:
[ERROR] /Users/bsolar/Projects/java-debug/com.microsoft.java.debug.plugin/src/main/java/com/microsoft/java/debug/plugin/internal/CompletionProposalRequestor.java:[324]
[ERROR]         return declaringType != null && org.eclipse.jdt.ls.core.internal.contentassist.TypeFilter.isFiltered(declaringType);
[ERROR]                                         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
[ERROR] org.eclipse.jdt.ls.core.internal.contentassist.TypeFilter cannot be resolved to a type
[ERROR] 1 problem (1 error)
```

--- a/com.microsoft.java.debug.plugin/src/main/java/com/microsoft/java/debug/plugin/internal/CompletionProposalRequestor.java
+++ b/com.microsoft.java.debug.plugin/src/main/java/com/microsoft/java/debug/plugin/internal/CompletionProposalRequestor.java
@@ -321,7 +321,7 @@ public final class CompletionProposalRequestor extends CompletionRequestor {
                 case CompletionProposal.JAVADOC_TYPE_REF:
                 case CompletionProposal.TYPE_REF: {
                     char[] declaringType = getDeclaringType(proposal);
-                    return declaringType != null && org.eclipse.jdt.ls.core.internal.contentassist.TypeFilter.isFiltered(declaringType);
+                    return declaringType != null && isFiltered(declaringType);
                 }
                 default: // do nothing
             }
@@ -330,6 +330,10 @@ public final class CompletionProposalRequestor extends CompletionRequestor {
         }
 
         return false;
+    }
+
+    private static boolean isFiltered(char[] fullTypeName) {
+        return JavaLanguageServerPlugin.getInstance().getTypeFilter().filter(new String(fullTypeName));
     }
 
     /**

--- a/com.microsoft.java.debug.plugin/src/main/java/com/microsoft/java/debug/plugin/internal/CompletionProposalRequestor.java
+++ b/com.microsoft.java.debug.plugin/src/main/java/com/microsoft/java/debug/plugin/internal/CompletionProposalRequestor.java
@@ -72,6 +72,8 @@ public final class CompletionProposalRequestor extends CompletionRequestor {
             CompletionItemKind.Text);
     // @formatter:on
 
+    private static boolean isFilterFailed = false;
+
     /**
      * Constructor.
      * @param typeRoot ITypeRoot
@@ -333,7 +335,6 @@ public final class CompletionProposalRequestor extends CompletionRequestor {
     }
 
     // Temp workaround for the completion error https://github.com/microsoft/java-debug/issues/534
-    private static boolean isFilterFailed = false;
     private static boolean isFiltered(char[] fullTypeName) {
         if (isFilterFailed) {
             return false;

--- a/com.microsoft.java.debug.plugin/src/main/java/com/microsoft/java/debug/plugin/internal/CompletionProposalRequestor.java
+++ b/com.microsoft.java.debug.plugin/src/main/java/com/microsoft/java/debug/plugin/internal/CompletionProposalRequestor.java
@@ -332,8 +332,21 @@ public final class CompletionProposalRequestor extends CompletionRequestor {
         return false;
     }
 
+    // Temp workaround for the completion error https://github.com/microsoft/java-debug/issues/534
+    private static boolean isFilterFailed = false;
     private static boolean isFiltered(char[] fullTypeName) {
-        return JavaLanguageServerPlugin.getInstance().getTypeFilter().filter(new String(fullTypeName));
+        if (isFilterFailed) {
+            return false;
+        }
+
+        try {
+            return JavaLanguageServerPlugin.getInstance().getTypeFilter().filter(new String(fullTypeName));
+        } catch (NoSuchMethodError ex) {
+            isFilterFailed = true;
+            JavaLanguageServerPlugin.logException("isFiltered for the completion failed.", ex);
+        }
+
+        return false;
     }
 
     /**

--- a/com.microsoft.java.debug.target/com.microsoft.java.debug.tp.target
+++ b/com.microsoft.java.debug.target/com.microsoft.java.debug.tp.target
@@ -15,7 +15,10 @@
             <unit id="org.eclipse.equinox.p2.core.feature.source.feature.group" version="0.0.0"/>
             <unit id="org.eclipse.equinox.sdk.feature.group" version="0.0.0"/>
             <unit id="org.eclipse.jdt.source.feature.group" version="0.0.0"/>
-            <unit id="org.eclipse.sdk.feature.group" version="0.0.0"/>
+            <unit id="org.eclipse.jdt.apt.pluggable.core" version="0.0.0"/>
+            <repository location="https://download.eclipse.org/eclipse/updates/4.31-I-builds/I20240130-1800/"/>
+        </location>
+        <location includeAllPlatforms="false" includeConfigurePhase="false" includeMode="planner" includeSource="true" type="InstallableUnit">
             <unit id="org.eclipse.xtext.sdk.feature.group" version="0.0.0"/>
             <repository location="https://download.eclipse.org/releases/2023-12/"/>
         </location>

--- a/pom.xml
+++ b/pom.xml
@@ -11,7 +11,7 @@
     <properties>
         <base.name>Java Debug Server for Visual Studio Code</base.name>
         <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
-        <tycho-version>4.0.3</tycho-version>
+        <tycho-version>4.0.5</tycho-version>
         <checkstyleDir>${basedir}</checkstyleDir>
     </properties>
 

